$NetBSD: patch-lib_src_libnyquist_nyquist_xlisp_xlisp_h,v 1.1 2013/01/08 22:37:27 makoto Exp $

nyquist/xlisp/xlfio.c:22:15: error: expected '=', ',', ';', 'asm' or '__attribute__' before 'error'
nyquist/xlisp/xlfio.c: In function 'xgetstroutput':

--- audacity-src-2.0.2/lib-src/libnyquist/nyquist/xlisp/xlisp.h.orig	2012-08-21 17:47:26.000000000 +0900
+++ lib-src/libnyquist/nyquist/xlisp/xlisp.h	2013-01-08 23:40:08.000000000 +0900
@@ -152,6 +152,16 @@ extern long ptrtoabs();
 #endif
 #endif
 
+/* NetBSD */
+#if defined(__NetBSD__) || defined(__DragonFlyBSD__)
+#include <sys/endian.h>
+#if BYTE_ORDER == LITTLE_ENDIAN
+#define XL_LITTLE_ENDIAN
+#else
+#define XL_BIG_ENDIAN
+#endif
+#endif
+
 /* Apple CC */
 #ifdef __APPLE__
 #define NNODES 2000
